src/app/person/person.component.ts

```diff
@@ -9,38 +9,45 @@ import { PersonService } from "src/app/services/person.component.service";
   styleUrls: ["./person.component.scss"]
 })
 export class PersonComponent implements OnInit {
-  constructor(public router: Router, private PersonService: PersonService) {}
+  personList: Array<object> = this.personService
+    .getPersons()
+    .sort((a: any, b: any) => a.id - b.id);
+  constructor(public router: Router, private personService: PersonService) {}
 
   ngOnInit() {}
 
-  return() {
+  navigateBack() {
     this.router.navigate(["todo"]);
   }
 
   //--------- CRUD Operations -----------
 
   addPersonFromService() {
-    this.PersonService.addPerson({
-      id: this.personList.length + 1,
+    this.personService.addPerson({
+      id: this.personService.nextId,
       name: "hagay"
     });
+    this.personService.getPersons().sort((a: any, b: any) => a.id - b.id);
```

Comment:
this is not the place to sort. if you think that soring after adding new person is required you should do it in the personService in the addPerson. 
Anyway, my approach is to sort only when you are about to display it. The list don't need to be sorted in the memory.